Improved layout and new color scheme

ajvarchetti opened the pull request:
Improved layout and feel. Also further implemented new color scheme that Erik created. I'm a huge fan of the way light mode looks. I don't feel the same with dark mode though. Still feel like the needs some more work.

jauligian committed:
Sizing/Spacing changes and set restriction on topic lengths.
Addition of New Theory button to main portion after generating thoery(WIP)

jauligian committed:
Second new theory button fixed and more spacing/alignment changes

ajvarchetti committed:
Improved layout and feel. Also further implemented new color scheme

jauligian committed:
Updates colors for dark mode.

jauligian committed:
Brought in some changes that were made in the FrontEnd branch and resolved some conflicts

jauligian committed:
Changed coloring of buttons and made them more consistent in styling.

jauligian commented:
Brought work I had done in a separate branch and tried to improve the look for dark mode to help fit the color scheme.

jauligian committed:
Uses browser local storage to save previously generated responses and added button to clear history. No confirmation option for clearing but may be something to add.

Changed new theory button in the sidebar to always be below the last item in the history, don't love it but was having issues having that and the clear history button at the bottom of the sidebar.

ajvarchetti commented:
Making a new PR
